extensions/unsplash/src/functions/copyFileToClipboard.ts

```diff
@@ -0,0 +1,53 @@
+import { showToast, ToastStyle, environment, getPreferenceValues } from "@raycast/api";
+import { runAppleScript } from "run-applescript";
+import { existsSync } from "fs";
+import { homedir } from "os";
+import { join } from "path";
+
+interface CopyFileToClipboardProps {
+  url: string;
+  id: string;
+}
+
+export const copyFileToClipboard = async ({ url, id }: CopyFileToClipboardProps) => {
+  const toast = await showToast(ToastStyle.Animated, "Downloading and copying image...");
+
+  const { path } = getPreferenceValues();
+  const selectedPath = path || environment.supportPath;
+  const fixedPathName = selectedPath.endsWith("/") ? `${selectedPath}${id}.jpg` : `${selectedPath}/${id}.jpg`;
+
+  try {
+    const actualPath = path ? join(homedir(), fixedPathName) : fixedPathName;
+
+    const command = !existsSync(actualPath)
+      ? `set cmd to "curl -o " & q_temp_folder & " " & "${url}"
+        do shell script cmd`
+      : "";
```

Comment:
Interesting that you use curl to download the file. You could also use Node, but well up to you.

Reply:
I hate the putting promises, callbacks and so much nested objects and so I prefer just a curl command 😅